ROLLO Website (#53)
arfc/rollo: docs/source/motivation.rst

.. _motivation:

.. image:: pics/rollo-logo.png
  :width: 450
  :alt: Alternative text

===================
Motivation
===================

> The evolutionary algorithm setup is highly customizable with an assortment of 

Reviewer: ```suggestion
Evolutionary algorithm setup is highly customizable with an assortment of 
```

genetic algorithm designs and operators. 
A reactor designer unfamiliar with evolutionary algorithms will have to go through the 
cumbersome process of customizing a genetic algorithm for their needs and determine 
which operators and hyperparameters work best for their problem.
Furthermore, computing fitness values with nuclear software is computationally 
expensive, necessitating using supercomputers.
Reactor designers have to set up parallelization to use the genetic algorithm 
optimization with nuclear software.

> The motivation for **ROLLO** is to provide a user-friendly tool that helps reactor designers use evolutionary algorithms for reactor design optimization.

Reviewer: Make the sentence on line 20 the first sentence in the Motivation section.

Author: Done

.. _motivation:

.. image:: pics/rollo-logo.png
  :width: 450
  :alt: Alternative text

===================
Motivation
===================
Previously, nuclear reactor designers individually customized available evolutionary algorithm 

> packages for their reactor design optimization problems.

Reviewer: Previously to what? This is a bit vague, consider adding some citations or extra descriptions of previous efforts.

Author: I removed it instead.

However, the evolutionary algorithm setup is highly customizable with an assortment of 
genetic algorithm designs and operators. 
A reactor designer unfamiliar with evolutionary algorithms will have to go through the 
cumbersome process of customizing a genetic algorithm for their needs and determine 
which operators and hyperparameters work best for their problem.
Furthermore, computing fitness values with nuclear software is computationally 
expensive, necessitating using supercomputers.
Reactor designers have to set up parallelization to use the genetic algorithm 
optimization with nuclear software.

Therefore, the motivation for **ROLLO** is to limit these inconveniences and facilitate

> the use of evolutionary algorithms for reactor design optimization.

Reviewer: Move this to the top of the motivation section

> Therefore, the motivation for **ROLLO** is to limit these inconveniences and facilitate
> the use of evolutionary algorithms for reactor design optimization.

Reviewer: ```suggestion
The motivation for **ROLLO** is to provide a user-friendly tool that helps reactor designers use evolutionary algorithms for reactor design optimization.
```

.. _motivation:

.. image:: pics/rollo-logo.png
  :width: 450
  :alt: Alternative text

===================
Motivation
===================

> Evolutionary algorithms have been applied to nuclear reactor design optimization. 

Reviewer 2: ```suggestion
```

> Previously, reactor designers individually customized available evolutionary algorithm 

Reviewer 2: ```suggestion
Previously, nuclear reactor designers individually customized available evolutionary algorithm 
```

packages for their reactor design optimization problems.
However, the evolutionary algorithm setup is highly customizable with an assortment of 
genetic algorithm designs and operators. 
A reactor designer unfamiliar with evolutionary algorithms will have to go through the 
cumbersome process of customizing a genetic algorithm for their needs and determine 
which operators and hyperparameters work best for their problem.
Furthermore, computing fitness values with nuclear software is computationally 
expensive, necessitating using supercomputers.
Reactor designers have to set up parallelization to use the genetic algorithm 
optimization with nuclear software.

Therefore, the motivation for **ROLLO** is to limit these inconveniences and facilitate

> using evolutionary algorithms for reactor design optimization.

Reviewer 2: ```suggestion
the use of evolutionary algorithms for reactor design optimization.
```

**ROLLO** provides a general genetic algorithm framework, sets up parallelization for the 
user, and promotes usability with an input file that only exposes mandatory parameters.
**ROLLO** strives to be effective, flexible, open-source, parallel, reproducible, and usable:

- Effective: **ROLLO** is well documented, tested, and version-controlled on `Github <https://github.com/arfc/rollo/>`_. 
- Flexible: **ROLLO** is nuclear software agnostic. Users can vary any imaginable parameter with any nuclear software. **ROLLO** uses a templating method to edit the input file of the coupled software.

> - Parallel: Users have the two options for running ROLLO in parallel.

Reviewer 2: ```suggestion
- Parallel: Users have two options for running **ROLLO** in parallel.
```

> - Reproducible: Data from every ROLLO run saves into a unique `pickled file <https://docs.python.org/3/library/pickle.html>`_.

Reviewer 2: ```suggestion
- Reproducible: Data from every **ROLLO** run saves into a unique `pickled file <https://docs.python.org/3/library/pickle.html>`_.
```
